### ISSUE TITLE
Support customizable cursor styles (block, line, block line, underline, etc)

### ISSUE THREAD

**Issue by ctrly4sh** [zed-industries/zed#16451]
### Check for existing issues

- [X] Completed

### Describe the feature

as far as i checked ZED only has line type cursor , but not block and underline one , the only way to use the block cursor is by enabling vim mode , which is kinda off , there should be a way to use the block cursor without the vim mode like the way it is there in vs code.
![Screenshot from 2024-08-19 12-59-43](https://github.com/user-attachments/assets/8ead0584-27b9-4a9c-b65d-d5d0175040e4)

**Comment by everdrone** [comment]
Pretty sure the only way to get a block cursor is to use vim mode right now

**Comment by ctrly4sh** [comment]
dont ya think it should be there ?

**Comment by everdrone** [comment]
I would like this feature too

**Comment by ctrly4sh** [comment]
bro i want your readme , i wanna make mine like of yours ? can you send it over or i should fork it ?

**Comment by everdrone** [comment]
@ctrly4sh feel free to fork or copy https://github.com/everdrone/everdrone/blob/master/readme.md?plain=1

**Comment by ctrly4sh** [comment]
thanks mate , do you have your discord , twitter  , IG or  something , kindly drop it down below
